enclave_build/src/cache.rs

```diff
@@ -0,0 +1,784 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// S&PDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::HashMap,
+    fs::{self, File, OpenOptions},
+    io::{self, Read, Write},
+    path::{Path, PathBuf},
+};
+
+use sha2::Digest;
+
+use crate::image::{self, ImageDetails};
+
+use oci_distribution::client::ImageData;
+
+/// Root folder for the cache
+pub const CACHE_ROOT_FOLDER: &str = "XDG_DATA_HOME";
+/// Name of the cache index file which stores the (image URI <-> image hash) mappings
+pub const CACHE_INDEX_FILE_NAME: &str = "index.json";
+/// The name of the folder used by the cache to store the image layers
+pub const CACHE_LAYERS_FOLDER_NAME: &str = "layers";
+/// The name of the image config file from the cache
+pub const CACHE_CONFIG_FILE_NAME: &str = "config.json";
+/// The name of the image manifest file from the cache
+pub const CACHE_MANIFEST_FILE_NAME: &str = "manifest.json";
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
I tried to make the errors as specific as possible.

Reply:
They can be specific, but right now there is some overlap between all the error enums. Generally crates expose a single Error-type / Result-type, so that someone who calls on the crate can implement a more unified way of dealing with them.